### DISCUSSION TITLE
"no peers" fallback strategies

### DISCUSSION THREAD

**Discussion by AskAlexSharov** [anacrolix/torrent#866] [Q&A] [answered]
Hello.
We have living network of peers for some data. It's self-sustained. 
But we would like to create some "fallback" service: if some file has no peers - node can download file from our service. 

What options are possible here? 
- we can't use existing public seedbox services - because their antivirus destroying our files (because our project is related to crypto-currency area).  
- can it be S3 http? or it should be BitTorrent protocol? 
- or just create our own seedbox? (better don't use cloud?)
- Do you see another options?

**Comment by anacrolix** [comment]
The easiest option is webseeding. That's just exposing the torrent files in the right way on a HTTP endpoint. This is well supported in anacrolix/torrent. Add a prefix URL and it will append the info name and file path. There's a BEP that describes the layout.

**Reply by AskAlexSharov** [reply, marked as answer]
thank you! i did know about such fallback existence - but was not able to find it this morning. 
here we go: https://www.bittorrent.org/beps/bep_0019.html - in lib it's `MetaInfo.UrlList`

**Reply by anacrolix** [reply]
Do you think some docs would help here? There is also https://pkg.go.dev/github.com/anacrolix/torrent#Torrent.AddWebSeeds.

**Reply by AskAlexSharov** [reply]
Yes, because `WebSeed` - it's bittorrent-specific termin. If I don't know `WebSeed` word - I can't find this feature in sources (I did try to search by http/httpClient/fallback/etc...). Even in bep_19 word `WebSeed` is not used (only in title) - so, it just doesn't stick to my head.

I would advise just add a bit words to `MetaInfo.UrlList` field comment.
